app/code/Magento/InventoryGroupedProductAdminUi/Plugin/Ui/DataProvider/Product/Form/AddColumnQuantityPerSource.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryGroupedProductAdminUi\Plugin\Ui\DataProvider\Product\Form;
+
+use Magento\GroupedProduct\Ui\DataProvider\Product\GroupedProductDataProvider;
+use Magento\InventoryCatalogApi\Model\IsSingleSourceModeInterface;
+use Magento\Ui\Component\Form\Element\DataType\Text;
+use Magento\Ui\Component\Listing\Columns\Column;
+
+/**
+ * In multi source mode add column "Quantity Per Source".
+ */
+class AddColumnQuantityPerSource
```

Comment:
Looks like this class needed only for the redundant observer
So need to remove this class and all of the familiar classes related to observer

Reply:
@naydav I think we doesn't need delete it class because it shows quantity on products wich we can assign to grouped or wich already assigned